### PR TITLE
⬆️ Prepare for Node 20 support

### PR DESCRIPTION
**Pull request by stoe**
This pull request includes updates to the CodeQL configuration and workflow files to enhance security scanning and improve CI/CD processes. The most important changes include adding a new CodeQL configuration file, updating the CodeQL workflow to use this configuration, and updating actions in the test workflow to specific versions.

CodeQL configuration and workflow updates:

* [`.github/codeql/codeql-config.yml`](diffhunk://#diff-3c168dbb160e6a97e58e9babfc08dcd60f89cf2c81ff592398940c29e576b870R1-R10): Added a new CodeQL configuration file to specify security-extended queries and paths to include and ignore.
* [`.github/workflows/codeql.yml`](diffhunk://#diff-12783128521e452af0cfac94b99b8d250413c516ec71fe6d97dbea666ff7ba27L18-L22): Updated the CodeQL workflow to remove unnecessary environment variables and to use the newly added CodeQL configuration file. [[1]](diffhunk://#diff-12783128521e452af0cfac94b99b8d250413c516ec71fe6d97dbea666ff7ba27L18-L22) [[2]](diffhunk://#diff-12783128521e452af0cfac94b99b8d250413c516ec71fe6d97dbea666ff7ba27R28)

GitHub Actions updates:

* [`.github/workflows/test.yml`](diffhunk://#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88L29-R31): Updated the versions of `actions/checkout` and `actions/setup-node` to specific commit SHAs for improved stability and security.

### CODE REVIEW

**Comment by stoe** [comment]
The one "error" we got from CodeQL is fine to ignore: https://github.com/ActionsDesk/ghec-invitations-report-action/security/code-scanning/tools/CodeQL/status/

**Review by stoe — COMMENTED** [review]
The one "error" we got from CodeQL is fine to ignore: https://github.com/ActionsDesk/ghec-invitations-report-action/security/code-scanning/tools/CodeQL/status/

File: action.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ outputs:
   base_sha:
     description: 'Report base SHA'
   head_sha:
-    description: 'Report head SHA'
+     description: 'Report head SHA'
```
Comment (gclhub) [inline action.yml:20]:
```suggestion
    description: 'Report head SHA'
```

Would that inserted space break the YAML formatting?

**Review by gclhub — APPROVED** [review]
Feel free the ignore the code suggestion - wasn't sure if that was intentional. Otherwise LGTM :shipit:

**Pull request merged**